File: src/test/java/com/github/vbauer/avconv4java/common/TestUtils.java
package com.github.vbauer.avconv4java.common;

import com.pushtorefresh.private_constructor_checker.PrivateConstructorChecker;

/**
 * @author Vladislav Bauer
 */

public final class TestUtils {

    private TestUtils() {
        throw new UnsupportedOperationException();
    }


    public static void checkUtilClassConstructor(final Class<?> utilsClass) {
        PrivateConstructorChecker.forClass(utilsClass)
            .expectedTypeOfException(UnsupportedOperationException.class)
            .check();
    }

}

File: src/test/kotlin/com/github/vbauer/avconv4java/common/TestUtils.kt
package com.github.vbauer.avconv4java.common

import com.pushtorefresh.private_constructor_checker.PrivateConstructorChecker

/**
 * @author Vladislav Bauer
 */

class TestUtils private constructor() {

    companion object {
        fun checkUtilClassConstructor(utilsClass: Class<*>) {
            PrivateConstructorChecker.forClass(utilsClass)
                    .expectedTypeOfException(UnsupportedOperationException::class.java)
                    .check()
        }
    }

}
